Speculative implementation of Registrar CSV processing

matt-bernhardt opened the pull request:
This implements the basic framework around an asynchronous worker that will handle the data import for records in the Registrar CSV file.

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [ ] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

JPrevost commented:
@matt-bernhardt I'll do a more thorough review on Monday, but initially this looks really good.

I didn't notice that if I access `/harvest` without no graduation lists previously loaded it errors so we'll want a sanity check to make sure that case is handled.

JPrevost reviewed (approved):
I suspect we'll adjust this a bit as we continue with the actual loading tasks, but this is a solid place to build from 👍🏻

matt-bernhardt commented:
I've pushed two additional commits to address your feedback:
- There was a missing "empty" template for the condition of having no uploaded registrar data. I added one.
- I removed the id pattern restriction on the `/harvest` path.

Seeing your approval, I'll merge now.
